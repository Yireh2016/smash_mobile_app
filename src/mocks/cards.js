export default [
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '2267',
    current_balance: 500,
    credit_utilization_percentage: 10,
    statement_balance: 2650,
    purchase_apr_percentage: 19,
    min_payment: 65,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 19,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 3100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9376',
    current_balance: 2000,
    credit_utilization_percentage: 25,
    statement_balance: 1030,
    purchase_apr_percentage: 21,
    min_payment: 37,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 21,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9085',
    current_balance: 3000,
    credit_utilization_percentage: 10,
    statement_balance: 1820,
    purchase_apr_percentage: 22,
    min_payment: 42,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 22,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1900,
      },
    ],
  },
];
