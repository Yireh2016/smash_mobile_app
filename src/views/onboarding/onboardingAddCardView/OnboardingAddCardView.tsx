/* eslint-disable react-native/no-inline-styles */
import * as React from 'react';
import { TouchableOpacity, View } from 'react-native';
import InfoLayout from '../../../layouts/infoLayout/InfoLayout';

import { MainParagraph, P1 } from '../../../components/texts/Texts';

import StepsProgressBar from '../../../components/stepsProgressBar/StepsProgressBar';

import ActionButton from '../../../components/actionButton/ActionButton';
import { Card } from '../../../types/Card';
import TotalDebt from '../../../components/totalDebt/TotalDebt';
import AddCardsTable from '../../../components/addCardsTable/AddCardsTable';
import formatMoneyValue from '../../../helpers/formatMoneyValue/formatMoneyValue';
import theme from '../../../theme/theme';
import PlaidBanksLinksViewWrapper, {
  PlaidLinkOnSuccessCallbackType,
} from '../../plaidBanksLinksView/PlaidBanksLinksViewWrapper';

interface AddCardProps {
  onNextPress: () => void;
  onSecurityPress: () => void;
  goToSafeAccountConnect: () => void;
  cards: Card[] | null;
  totalDebt: number;
  onAddCardSuccess: PlaidLinkOnSuccessCallbackType;
}

const AddCardView: React.FunctionComponent<AddCardProps> = ({
  onNextPress,
  onSecurityPress,
  goToSafeAccountConnect,
  cards,
  totalDebt,
  onAddCardSuccess,
}) => {
  const footer = (
    <View
      style={{
        marginTop: 16,
        flexDirection: 'row',
        justifyContent: 'center',
      }}
    >
      <P1>Learn more about our </P1>
      <TouchableOpacity onPress={onSecurityPress}>
        <P1 isBold={true} color={theme.palette.secondary}>
          Security
        </P1>
      </TouchableOpacity>
    </View>
  );
  return (
    <InfoLayout onPress={onNextPress} actionLabel={'Next'} footer={footer}>
      <MainParagraph style={{ marginBottom: 16 }}>Steps 1 of 3</MainParagraph>
      <View style={{ marginBottom: 16 }}>
        <StepsProgressBar progress={100 / 3} />
      </View>
      <MainParagraph>
        Add all your credit cards to create your personalized plan
      </MainParagraph>
      {totalDebt > 0 && (
        <View style={{ marginBottom: 20 }}>
          <TotalDebt totalDebt={formatMoneyValue(totalDebt)} />
        </View>
      )}
      {cards && cards.length > 0 && (
        <View>
          <AddCardsTable cards={cards} />
        </View>
      )}
      <View style={{ marginTop: cards && cards?.length > 0 ? 16 : 20 }}>
        {cards && cards.length > 0 ? (
          <PlaidBanksLinksViewWrapper onSuccessCallback={onAddCardSuccess}>
            <ActionButton label="Add card" />
          </PlaidBanksLinksViewWrapper>
        ) : (
          <ActionButton label="Add card" onPress={goToSafeAccountConnect} />
        )}
      </View>
    </InfoLayout>
  );
};

export default AddCardView;
