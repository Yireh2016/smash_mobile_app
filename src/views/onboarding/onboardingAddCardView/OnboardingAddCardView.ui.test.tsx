import * as React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import renderer from 'react-test-renderer';
import cards from '../../../mocks/cards';
import OnboardingAddCardView from './OnboardingAddCardView';
const AddCardViewProps = {
  onNextPress: () => {},
  onAddCardSuccess: () => {},
  onSecurityPress: () => {},
  cards,
  totalDebt: 1232,
};

describe('', () => {
  it('AddCard onboarding view with cards renders correctly', () => {
    const Element = () => {
      return (
        <SafeAreaProvider>
          <OnboardingAddCardView {...AddCardViewProps} />
        </SafeAreaProvider>
      );
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('AddCard onboarding view without cards renders correctly', () => {
    const AddCardViewProps2 = JSON.parse(JSON.stringify(AddCardViewProps));
    AddCardViewProps2.cards = null;
    AddCardViewProps2.totalDebt = 0;
    const Element = () => {
      return (
        <SafeAreaProvider>
          <OnboardingAddCardView {...AddCardViewProps2} />
        </SafeAreaProvider>
      );
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
