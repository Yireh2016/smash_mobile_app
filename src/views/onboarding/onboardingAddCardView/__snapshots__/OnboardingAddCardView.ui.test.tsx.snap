// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCard onboarding view with cards renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`AddCard onboarding view without cards renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
